src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -1985,14 +1985,13 @@ void GameObject::ModifyHealth(int32 change, Unit* attackerOrHealer /*= NULL*/, u
     // dealing damage, send packet
     if (player)
     {
-        WorldPacket data(SMSG_DESTRUCTIBLE_BUILDING_DAMAGE, 8 + 8 + 8 + 4 + 4);
-        data << GetPackGUID();
-        data << attackerOrHealer->GetPackGUID();
-        data << player->GetPackGUID();
-        data << uint32(-change);                    // change  < 0 triggers SPELL_BUILDING_HEAL combat log event
-                                                    // change >= 0 triggers SPELL_BUILDING_DAMAGE event
-        data << uint32(spellId);
-        player->SendDirectMessage(&data);
+        WorldPackets::GameObject::DestructibleBuildingDamage packet;
+        packet.Caster = player->GetGUID();
+        packet.Target = GetGUID();
+        packet.Damage = -change;
+        packet.Owner = attackerOrHealer->GetGUID(); // todo: this can be a GameObject
```

Comment:
I took the structure from the WPP, and saw 
```
ServerToClient: SMSG_DESTRUCTIBLE_BUILDING_DAMAGE (0x02C8) Length: 45 ConnIdx: 1 Time: 08/03/2015 11:43:27.954 Number: 28452
Target: GameObject Map: 607 (Strand of the Ancients) Entry: 190724 (Gate of the Blue Sapphire)
Caster: Player 
Owner: GameObject Map: 607 (Strand of the Ancients) Entry: 190752 (Massive Seaforium Charge)
Damage: 4500
SpellID: 52408
```

I will swap Caster and Owner :)